Xamarin.Forms.Core/BindingExpression.cs

```diff
@@ -410,11 +406,57 @@ public BindingPair(BindingExpressionPart part, object source, bool isLast)
 			public object Source { get; private set; }
 		}
 
+		class WeakPropertyChangedProxy
+		{
```

Comment:
Instead of this, couldn't you reuse the https://github.com/xamarin/Xamarin.Forms/blob/master/Xamarin.Forms.Core/WeakEventManager.cs ?

I'm uneasy about having multiple classes doing very similar stuffs.